### PR TITLE
textFit accepts an array of HTMLElement

### PR DESCRIPTION
**Pull request by mu-hun**
According to the projects [README](https://github.com/STRML/textFit#usage) the textFit() function accepts either a single HTMLElement or HTMLElement[].

The added other union types `NodeListOf<Element> | HTMLCollection | null` referanced from [lasted commit of textFit repository](https://github.com/STRML/textFit/commit/5f8a94#diff-7aa4473ede4abd9ec099e87fec67fd57afafaf39e05d493ab4533acc38547eb8R20).

I've been thinking about removing this type definiton to DefinitelyTyped. However, this package latest version was released 3 year ago without type definition file. Therefore, I update this definition.

---

This issue started from https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/58703 discussion.

Co-authored-by: Marijn Huizendveld <marijn@huizendveld.com>

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by x86chi** [comment]
```
~/github/DefinitelyTyped/DefinitelyTyped/types/textfit patch-1*
❯ npm test textFit

> definitely-typed@0.0.3 test
> dtslint types "textFit"

Error: ENOENT: no such file or directory, open '../notNeededPackages.json'
```

I tried to run test for current changes. but I saw uppper log 😢 How can I fix it?

File: types/textfit/textfit-tests.ts [hunk 1]
```diff
@@ -14,3 +15,15 @@ const options: textFit.TextFitOption = {
 };
 
 textFit(element, options);
+textFit(elements, options);
+
+const ul = document.createElement('ul');
+ul.append(document.createElement('li'));
+
+const nodeList = ul.querySelectorAll('li');
```
Comment (marijn) [inline types/textfit/textfit-tests.ts:23]: I'm not really confident that `NodeList` is supported, I wasn't able to find this as an example in the library, though I may have overlooked it. Where did you see it referenced?

**Review by marijn — COMMENTED** [review]
Thanks for making this PR. I wasn't able at the time of filing the issue 🙏

File: types/textfit/textfit-tests.ts (hunk 1 above)
Comment (x86chi) [reply types/textfit/textfit-tests.ts:23]: I referenced [`index.d.ts` of textFit repository](https://github.com/STRML/textFit/commit/5f8a94#diff-7aa4473ede4abd9ec099e87fec67fd57afafaf39e05d493ab4533acc38547eb8R20). 

This package already have own types a year ago, but is still work-in-progress status. 

please refer https://github.com/STRML/textFit/issues/60 issue.

**Comment by x86chi** [comment]
> ```
> ~/github/DefinitelyTyped/DefinitelyTyped/types/textfit patch-1*
> ❯ npm test textFit
> 
> > definitely-typed@0.0.3 test
> > dtslint types "textFit"
> 
> Error: ENOENT: no such file or directory, open '../notNeededPackages.json'
> ```
> 
> I tried to run test for current changes. but I saw uppper log 😢 How can I fix it?

Hello maintainers, Could you help from upper comment? I force "change to ready for review" to ask for help...

**Comment by joshuajeschek** [comment]
@x86chi could you try again using `npm test textfit`? Note that I spelled "textfit" all lowercase.
For me I got a different error than yours before, but `npm test textfit` worked without any problems.

**Comment by x86chi** [comment]
```
❯ npm test textfit

> definitely-typed@0.0.3 test
> dtslint types "textfit"

Error: ENOENT: no such file or directory, open '../notNeededPackages.json'
```

@joshuajeschek Thanks for comment.. but It occured same output :(

**Comment by joshuajeschek** [comment]
okay. you could also try updating the master branch to be up to date with DT, and then merge it into your branch. There was a recent update to `notNeededPackages.json`

**Comment by x86chi** [comment]
@joshuajeschek I'm still seeing the same output. 

> For me I got a different error than yours before, but npm test textfit worked without any problems.

You said there was nothing error in `notNeededPackages.json`, I wonder what other issues there were. Could you tell me more?

**Comment by joshuajeschek** [comment]
> You said there was nothing error in `notNeededPackages.json`, I wonder what other issues there were. Could you tell me more?

Oh yeah, here it is:
```terminal
❯ npm test textFit

> definitely-typed@0.0.3 test
> dtslint types "textFit"

dtslint@0.0.110
Error: C:/.../DefinitelyTyped/types/textFit/textfit-tests.ts:1:26
ERROR: 1:26  expect  TypeScript@4.7 compile error:
File name 'C:/.../DefinitelyTyped/types/textfit/index.d.ts' differs from already included file name 'C:/.../DefinitelyTyped/types/textFit/index.d.ts' only in casing.
  The file is in the program because:
    Root file specified for compilation
    Imported via 'textfit' from file 'C:/.../DefinitelyTyped/types/textFit/textfit-tests.ts'

    at testTypesVersion (C:\...\DefinitelyTyped\node_modules\@definitelytyped\dtslint\dist\index.js:194:15)
    at async runTests (C:\...\DefinitelyTyped\node_modules\@definitelytyped\dtslint\dist\index.js:170:13)
    at async main (C:\...\DefinitelyTyped\node_modules\@definitelytyped\dtslint\dist\index.js:79:9)
```

please note that this is with `textFit` not `textfit`. With `textfit`, everything works perfectly.

**Comment by joshuajeschek** [comment]
Since it works on my machine and it also works on Github Actions, I would suggest that you continue to without tests on you local machine and rely on the Github Actions to solve that.
Maybe your local clone is somehow corrupted, you could also try cloning [x86chi/DefinitelyTyped](https://github.com/x86chi/DefinitelyTyped) again.

File: types/textfit/index.d.ts [hunk 2]
```diff
@@ -3,7 +3,10 @@
 // Definitions by: Muhun Kim <https://github.com/x86chi>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-declare function textFit(element: HTMLElement, options?: textFit.TextFitOption): void;
+declare function textFit(
+    els: HTMLElement | HTMLElement[] | NodeListOf<Element> | HTMLCollection | null,
```
Comment (peterblazejewicz) [inline types/textfit/index.d.ts:7]: this should also accept jQuery object, should it be supported here?
`els` cannot be a null at any point in the current implementation, there is not nullable expression kind of check at any point:
https://github.com/STRML/textFit/blob/master/textFit.js#L61

**Comment by x86chi** [comment]
@joshuajeschek When I cloned and tested again, there was no `dtslint` command. so I ran it again **with master-based dependencies** the test passed. please refer below terminal log:

```shell
~/github/DefinitelyTyped/DefinitelyTyped patch-1* 9s
❯ npm test textFit

> definitely-typed@0.0.3 test
> dtslint types "textFit"

sh: dtslint: command not found

~/github/DefinitelyTyped/DefinitelyTyped patch-1*
❯ git switch master
Updating files: 100% (29826/29826), done.
Switched to branch 'master'
Your branch is up to date with 'origin/master'.

~/github/DefinitelyTyped/DefinitelyTyped master* 12s
❯ npm i           

added 37 packages, removed 83 packages, changed 1 package, and audited 511 packages in 3s

58 packages are looking for funding
  run `npm fund` for details

6 high severity vulnerabilities

To address all issues possible (including breaking changes), run:
  npm audit fix --force

Some issues need review, and may require choosing
a different dependency.

Run `npm audit` for details.

~/github/DefinitelyTyped/DefinitelyTyped master* 6s
❯ git switch patch-1     
Updating files: 100% (29826/29826), done.
Switched to branch 'patch-1'
Your branch is up to date with 'origin/patch-1'.

~/github/DefinitelyTyped/DefinitelyTyped patch-1* 13s
❯ npm test textfit              

> definitely-typed@0.0.3 test
> dtslint types "textfit"

dtslint@0.0.111
```

File: types/textfit/index.d.ts (hunk 2 above)
Comment (x86chi) [reply types/textfit/index.d.ts]: Thanks for review with suggestion! Your suggestion has been apply to new commit. Can you review again?

**Comment by joshuajeschek** [comment]
@x86chi after doing `npm i`, dtslint was back, wasn't it? I think you already fixed it! 😀

**Comment by x86chi** [comment]
@joshuajeschek Sadly in my local machine, still dtslint command not found with `path-1` branch based dependencies 😢

```
~/github/DefinitelyTyped/DefinitelyTyped patch-1* 12s
❯ npm i           

added 83 packages, removed 37 packages, and audited 557 packages in 9s

59 packages are looking for funding
  run `npm fund` for details

5 high severity vulnerabilities

To address all issues possible (including breaking changes), run:
  npm audit fix --force

Some issues need review, and may require choosing
a different dependency.

Run `npm audit` for details.

~/github/DefinitelyTyped/DefinitelyTyped patch-1* 13s
❯ npm test textfit

> definitely-typed@0.0.3 test
> dtslint types "textfit"

sh: dtslint: command not found
````

**Comment by joshuajeschek** [comment]
@x86chi then I also don't know 🤷

**Comment by x86chi** [comment]
My local test results are strange, but is it enough to approve this PR? @joshuajeschek

**Comment by joshuajeschek** [comment]
> My local test result is strange, but is it enough to approve this PR? @joshuajeschek

I don't know if @peterblazejewicz still has something to add, let's wait for him :)
Apart from that, CI results are what counts ^^

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Comment by x86chi** [comment]
```
Error: In package.json: Dependency @types/jquery not in the allowed dependencies list.
Don't use a 'package.json' for @types dependencies unless this package relies on
an old version of types that have since been moved to the source repo.
For example, if package *P* used to have types on Definitely Typed at @types/P,
but now has its own types, a dependent package *D* will need to use package.json
to refer to @types/P if it relies on old versions of P's types.
In this case, please make a pull request to microsoft/DefinitelyTyped-tools adding @types/P to `packages/definitions-parser/allowedPackageJsonDependencies.txt`.
```

> https://github.com/DefinitelyTyped/DefinitelyTyped/runs/5463172056?check_suite_focus=true#step:6:13

Guess I'll have to go contribute to https://github.com/microsoft/DefinitelyTyped-tools  for add `@types/jquery` to list 😕

**Comment by peterblazejewicz** [comment]
@x86chi `jquery` has no native TS code, so it's resolved to `@types/jquery` when using autoamtic types accquisition feature in TypeScript (and in the IDEs). 
It should work here, either as module import or reference:
```ts
import * as $ from 'jquery';
// or
/// <reference types="jquery" />
```

**Comment by x86chi** [comment]
@peterblazejewicz All right. I have applied your suggestion. Can you review again?

**Review by Grayson-code — APPROVED** [review]
seems like its fine now

**Comment by x86chi** [comment]
@peterblazejewicz Can you review again? 👀

**Review by peterblazejewicz — APPROVED** [review]

**Comment by x86chi** [comment]
Ready to merge!

**Pull request merged**